lib/linkify.dart

```diff
@@ -79,17 +98,23 @@ final _linkifyEmailRegex = RegExp(
   caseSensitive: false,
 );
 
+final _linkifyPhoneRegex = RegExp(
+  r"^((?:.|\n)*?)((tel:)?([+0-9]{2,4}?\s?[0-9\s\(\)]{8,15}))",
+  caseSensitive: false,
+);
+
 /// Turns [text] into a list of [LinkifyElement]
 ///
 /// Use [humanize] to remove http/https from the start of the URL shown.
 /// Will default to `false` (if `null`)
 ///
-/// Uses [linkTypes] to enable some types of links (URL, email).
+/// Uses [linkTypes] to enable some types of links (URL, email, phone).
 /// Will default to all (if `null`).
 List<LinkifyElement> linkify(
   String text, {
   bool humanize,
   List<LinkType> linkTypes,
+  RegExp phoneRegex,
```

Comment:
Instead of adding custom regexes here, let's implement it in a different PR with a more generic approach, so that we can parse anything (not just custom phones).

Reply:
Currently it gives the user a very quick way to highlight urls, emails, phones which are the most used cases.
So we can add another method which can handle generic parsing with regexs user provides.

Reply:
The point to add an optional phoneRegex here is that the library usage still is **isn't complicated for general use cases** while also providing support for phone numbers of different regions. 
So we can cover 3 levels of users:
1. Quick prototyping users which can accept our generic phone url
2. Users which would want to support only phone formats of their country.
For Next PR
3. Users who would want high degree of Fexibility ( implemented in a different method )

Reply:
I think having a proper custom parsing method is better than rushing one for a single type, and have to introduce breaking changes when we migrate it to the new way.

We'll implement a way to specify `RegExp` -> custom `LinkifyElement`, which will allow for custom phone, email, and link regexs, plus any custom element.

Reply:
Seems like this param is still here